src/pudl/analysis/allocate_net_gen.py

```diff
@@ -429,41 +528,46 @@ def standardize_input_frequency(
-    at the generator/fuel-type level (i.e. the result of :func:`allocate_gen_fuel_by_generator_energy_source`)
-    to be associated and scaled with all of the owners of those generators.
-    This allows the aggregation of fuel use to the utility level.
+    It can be helpful to have a table of net generation and fuel consumption at the
+    generator/fuel-type level (i.e. the result of
+    :func:`allocate_gen_fuel_by_generator_energy_source`) to be associated and scaled
+    with all of the owners of those generators.  This allows the aggregation of fuel use
+    to the utility level.
 
-    Scaling generators with their owners' ownership fraction is currently
-    possible via :class:`pudl.analysis.plant_parts_eia.MakeMegaGenTbl`. This
-    function uses the allocated net generation at the generator/fuel-type level,
-    merges that with a generators table to ensure all necessary columns are
-    available, and then feeds that table into the EIA Plant-parts' :meth:`scale_by_ownership`.
+    Scaling generators with their owners' ownership fraction is currently possible via
+    :class:`pudl.analysis.plant_parts_eia.MakeMegaGenTbl`. This function uses the
+    allocated net generation at the generator/fuel-type level, merges that with a
+    generators table to ensure all necessary columns are available, and then feeds that
+    table into the EIA Plant-parts' :meth:`scale_by_ownership`.
 
     Args:
-        gen_pm_fuel: able of allocated generation at the generator/prime mover/energy
-            source. Result of :func:`allocate_gen_fuel_by_generator_energy_source`
-        gens: `generators_eia860` table with cols: :const:``IDX_GENS``, `capacity_mw`
-            and `utility_id_eia`
-        own_eia860: `ownership_eia860` table.
+        net_gen_fuel_alloc: table of allocated generation at the generator, prime mover,
+            and energy source. From :func:`allocate_gen_fuel_by_generator_energy_source`
+        gens: ``generators_eia860`` table with cols: :const:``IDX_GENS``,
+            ``capacity_mw`` and ``utility_id_eia``
+        own_eia860: ``ownership_eia860`` table.
     """
-    gen_pm_fuel_own = pudl.analysis.plant_parts_eia.MakeMegaGenTbl().scale_by_ownership(
-        gens_mega=pd.merge(
-            gen_pm_fuel,
-            gens[IDX_GENS + ["utility_id_eia", "capacity_mw"]],
-            on=IDX_GENS,
-            validate="m:1",
+    return pudl.analysis.plant_parts_eia.MakeMegaGenTbl().scale_by_ownership(
+        gens_mega=pudl.helpers.date_merge(
+            left=net_gen_fuel_alloc,
+            right=gens[IDX_GENS + ["utility_id_eia", "capacity_mw"]],
+            left_date_col="report_date",
+            right_date_col="report_date",
+            new_date_col="report_date",
+            on=["plant_id_eia", "generator_id"],
+            date_on=["year"],
             how="left",
+            report_at_start=True,
+            validate="m:1",
```

Comment:
This seems like a weird entanglement with downstream analysis / PPE tooling. I guess `scale_by_ownership()` is a static method? Should it be pulled out of `MakeMegaGensTbl()` for re-use in other contexts?  Maybe this is something that's dealt with in the MCOE / PPE dagsterization PRs?

Reply:
As far as I can tell, this is the only time `MakeMegaGensTbl()` is used outside of the context of the PPE. @cmgosnell is this still a useful function? I wasn't sure how to deal with this in the PPE PR since mega gens is downstream of this.

Reply:
IIRC the ownership scaling was just a convenient piece of code to re-use that happened to be embedded within a class that had been defined for another purpose.

But I think the generation-fuel scaled by ownership is (currently) only used in the context of the PPE.  Do we think it's a generally useful / interesting output that should be published alongside the other generation fuel outputs? Or is it really just an input into the PPE? @cmgosnell

Reply:
Seems like this generation and fuel table by owner table can just be taken out?

Reply:
this isn't an input into the PPE. I don't think it feeds into anything else in PUDL rn. it uses a PPE method. which perhaps makes a case that this method is more broadly useful and shouldn't be in PPE land because I've also used this method in other rmi tasks. i don't think the use of this method in a different context should block moving forward on dagster stuff. it's been used over here for... idk years at this point. but moving it out into helpers or something would maybe be good as well.

this output was originally asked for by rmi folks and has subsequently been used by greg m as well iirc.